chain/jsonrpc/src/lib.rs

```diff
@@ -854,7 +804,170 @@ impl JsonRpcHandler {
+        near_jsonrpc_primitives::types::maintenance::RpcMaintenanceWindowsError,
+    > {
+        let near_jsonrpc_primitives::types::maintenance::RpcMaintenanceWindowsRequest {
+            account_id,
+        } = request;
+        let windows = self.view_client_send(GetMaintenanceWindows { account_id }).await?;
+        Ok(windows.iter().map(|r| (r.start, r.end)).collect())
+    }
+
+    async fn congestion_level(
+        &self,
+        request_data: near_jsonrpc_primitives::types::congestion::RpcCongestionLevelRequest,
+    ) -> Result<
+        near_jsonrpc_primitives::types::congestion::RpcCongestionLevelResponse,
+        near_jsonrpc_primitives::types::congestion::RpcCongestionLevelError,
+    > {
+        let chunk_view =
+            self.view_client_send(GetChunk::rpc_from(request_data.chunk_reference)).await?;
+        let config_result = self
+            .view_client_send(GetProtocolConfig(BlockReference::BlockId(BlockId::Height(
+                chunk_view.header.height_included,
+            ))))
+            .await;
+        let config = config_result.map_err(|err: RpcProtocolConfigError| match err {
+            RpcProtocolConfigError::UnknownBlock { error_message } => {
+                near_jsonrpc_primitives::types::congestion::RpcCongestionLevelError::UnknownBlock {
+                    error_message,
+                }
+            }
+            RpcProtocolConfigError::InternalError { error_message } => {
+                near_jsonrpc_primitives::types::congestion::RpcCongestionLevelError::InternalError {
+                    error_message,
+                }
+            }
+        })?;
+        let congestion_info = chunk_view.header.congestion_info;
+        let congestion_level = congestion_info
+            .map(|info| info.congestion_level(config.runtime_config.congestion_control_config))
+            .unwrap_or(0.0);
+        Ok(near_jsonrpc_primitives::types::congestion::RpcCongestionLevelResponse {
+            congestion_level,
+        })
+    }
+
+    async fn changes_in_block_by_type(
+        &self,
+        request: near_jsonrpc_primitives::types::changes::RpcStateChangesInBlockByTypeRequest,
+    ) -> Result<
+        near_jsonrpc_primitives::types::changes::RpcStateChangesInBlockResponse,
+        near_jsonrpc_primitives::types::changes::RpcStateChangesError,
+    > {
+        let block: near_primitives::views::BlockView =
+            self.view_client_send(GetBlock(request.block_reference)).await?;
+
+        let block_hash = block.header.hash;
+        let changes = self
+            .view_client_send(GetStateChanges {
+                block_hash,
+                state_changes_request: request.state_changes_request,
+            })
+            .await?;
+
+        Ok(near_jsonrpc_primitives::types::changes::RpcStateChangesInBlockResponse {
+            block_hash: block.header.hash,
+            changes,
+        })
+    }
+
+    async fn client_config(
```

Comment:
I wonder why you haven't included some public methods into a trait `JsonRpcHandlerExt`? Am I missing something? Are you missing something? :)

Reply:
Yep, I had a brain lag. Now I've double checked that all methods that are used in `process_basic_requests_internal` are available in trait